src/main/java/de/sventorben/keycloak/authentication/hidpd/HomeIdpDiscoveryAuthenticator.java

```diff
@@ -33,6 +33,11 @@ public void authenticate(AuthenticationFlowContext authenticationFlowContext) {
 
         if (context.loginPage().shouldByPass()) {
             String loginHint = trimToNull(context.loginHint().getFromSession());
+
+            final UserModel user = authenticationFlowContext.getAuthenticationSession().getAuthenticatedUser();
+            if (loginHint == null && user != null) {
+                loginHint = trimToNull(user.getEmail());
```

Comment:
This is hard coded to the email address, but needs to respect authenticator configuration. Try to use something like `user.getFirstAttribute(context.config().userAttribute())`.